src/Microsoft.Extensions.Configuration.Abstractions/IConfigurationProvider.cs

```diff
@@ -5,17 +5,38 @@
 
 namespace Microsoft.Extensions.Configuration
 {
+    /// <summary>
+    /// Represents a source of configuration key/values for an application.
+    /// </summary>
     public interface IConfigurationProvider
     {
+        /// <summary>
+        /// Tries to get a configuration value for the specified key.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <param name="value">The value.</param>
+        /// <returns><c>True</c> if a value for the specified key was found, otherwise <c>false</c>.</returns>
         bool TryGet(string key, out string value);
 
+        /// <summary>
+        /// Sets a configuration value for the specified key.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <param name="value">The value.</param>
         void Set(string key, string value);
 
+        /// <summary>
+        /// Loads configuration values from the source represented by this <see cref="IConfigurationProvider"/>.
+        /// </summary>
         void Load();
 
-        IEnumerable<string> GetChildKeys(
-            IEnumerable<string> earlierKeys,
-            string parentPath,
-            string delimiter);
+        /// <summary>
+        /// TODO: No idea what this is, help pls.
+        /// </summary>
+        /// <param name="earlierKeys"></param>
+        /// <param name="parentPath"></param>
+        /// <param name="delimiter"></param>
```

Comment:
```
The delimiter to use to identify keys in the <see cref="IConfigurationProvider"/>'s data.
```
BTW, I find it weird that the delimiter is part of the contract for this method. I would find it more natural for providers to completely abstract however they want to represent keys in their internal state from the consuming side.

cc @lodejard to make sure this is intentional rather than leftovers from a previous design iteration.